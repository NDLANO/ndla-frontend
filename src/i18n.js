/**
 * Copyright (c) 2018-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { formatNestedMessages } from '@ndla/i18n';
import { messagesNB, messagesEN, messagesNN } from '@ndla/ui';
import additionalMessagesNB from './messages/messagesNB';
import additionalMessagesNN from './messages/messagesNN';
import additionalMessagesEN from './messages/messagesEN';

const NB = {
  name: 'BokmÃ¥l',
  abbreviation: 'nb',
  messages: formatNestedMessages({ ...messagesNB, ...additionalMessagesNB }),
};
const NN = {
  name: 'Nynorsk',
  abbreviation: 'nn',
  messages: formatNestedMessages({ ...messagesNN, ...additionalMessagesNN }),
};
const EN = {
  name: 'English',
  abbreviation: 'en',
  messages: formatNestedMessages({ ...messagesEN, ...additionalMessagesEN }),
};

export const appLocales = [NB, NN, EN];
export const preferredLocales = [NB, NN];

export const getLocaleObject = localeAbbreviation => {
  const locale = appLocales.find(l => l.abbreviation === localeAbbreviation);

  return locale || NB; // defaults to NB
};

export const isValidLocale = localeAbbreviation =>
  appLocales.find(l => l.abbreviation === localeAbbreviation) !== undefined;

export const getHtmlLang = localeAbbreviation => {
  const locale = appLocales.find(l => l.abbreviation === localeAbbreviation);
  return locale ? locale.abbreviation : 'nb'; // Defaults to nb if not found
};

export function getLocaleInfoFromPath(path) {
  const paths = path.split('/');
  const basename = isValidLocale(paths[1]) ? paths[1] : '';
  const basepath = basename ? path.replace(`/${basename}`, '') : path;
  return {
    basepath,
    basename,
    ...getLocaleObject(basename),
  };
}
