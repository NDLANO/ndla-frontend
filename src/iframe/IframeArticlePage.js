/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';

import { OneColumn, CreatedBy } from '@ndla/ui';
import { withTracker } from '@ndla/tracker';
import { withTranslation } from 'react-i18next';
import { transformArticle } from '../util/transformArticle';
import Article from '../components/Article';
import { getArticleScripts } from '../util/getArticleScripts';
import { ResourceShape, ArticleShape, LocationShape } from '../shapes';
import { getArticleProps } from '../util/getArticleProps';
import { getAllDimensions } from '../util/trackingUtil';
import PostResizeMessage from './PostResizeMessage';
import FixDialogPosition from './FixDialogPosition';
import { SocialMediaMetadata } from '../components/SocialMediaMetadata';
import { fetchResource } from '../containers/Resources/resourceApi';
import config from '../config';

export const fetchResourceId = props => {
  const paths = props.location.pathname.split('/');
  return paths.find(path => path.startsWith('urn'));
};

class IframeArticlePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      path: undefined,
    };
  }

  componentDidMount() {
    fetchResource(fetchResourceId(this.props), this.props.locale)
      .then(resource => {
        this.setState({
          path: resource.path || resource.paths?.[0],
        });
      })
      .catch(error => {});
  }

  static willTrackPageView(trackPageView, currentProps) {
    const { resource } = currentProps;
    if (resource?.article?.id) {
      trackPageView(currentProps);
    }
  }

  static getDimensions(props) {
    const articleProps = getArticleProps(props.resource);
    const {
      resource: { article },
    } = props;
    return getAllDimensions({ article }, articleProps.label, true);
  }

  static getDocumentTitle({ t, resource }) {
    if (resource?.article?.id) {
      return `NDLA | ${resource.article.title}`;
    }
    return '';
  }
  render() {
    const { resource, locale, location, t } = this.props;
    const article = transformArticle(this.props.article, locale);
    const scripts = getArticleScripts(article);
    const contentUrl = this.state.path
      ? `${config.ndlaFrontendDomain}${this.state.path}`
      : undefined;
    return (
      <OneColumn>
        <Helmet>
          <title>{`NDLA | ${article.title}`}</title>
          {scripts.map(script => (
            <script
              key={script.src}
              src={script.src}
              type={script.type}
              async={script.async}
              defer={script.defer}
            />
          ))}
        </Helmet>
        <SocialMediaMetadata
          title={article.title}
          location={location}
          image={article.metaImage}
          description={article.metaDescription}
          locale={locale}
          trackableContent={article}
        />
        <PostResizeMessage />
        <FixDialogPosition />
        <Article
          article={article}
          locale={locale}
          modifier="clean iframe"
          {...getArticleProps(resource)}>
          <CreatedBy
            name={t('createdBy.content')}
            description={t('createdBy.text')}
            url={contentUrl}
          />
        </Article>
      </OneColumn>
    );
  }
}

IframeArticlePage.propTypes = {
  locale: PropTypes.string.isRequired,
  resource: ResourceShape,
  article: ArticleShape,
  location: LocationShape,
};

export default withTranslation()(withTracker(IframeArticlePage));
