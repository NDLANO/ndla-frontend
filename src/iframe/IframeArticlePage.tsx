/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { Helmet } from 'react-helmet';

import { OneColumn, CreatedBy } from '@ndla/ui';
import { withTracker } from '@ndla/tracker';
import { CustomWithTranslation, withTranslation } from 'react-i18next';
import { transformArticle } from '../util/transformArticle';
import Article from '../components/Article';
import { getArticleScripts } from '../util/getArticleScripts';
import { getArticleProps } from '../util/getArticleProps';
import { getAllDimensions } from '../util/trackingUtil';
import PostResizeMessage from './PostResizeMessage';
import FixDialogPosition from './FixDialogPosition';
import SocialMediaMetadata from '../components/SocialMediaMetadata';
import config from '../config';
import { LocaleType } from '../interfaces';
import { GQLArticle, GQLResource } from '../graphqlTypes';

interface Props extends CustomWithTranslation {
  locale?: LocaleType;
  resource?: GQLResource;
  article: GQLArticle;
}

const IframeArticlePage = ({
  resource,
  locale: propsLocale,
  t,
  article: propsArticle,
  i18n,
}: Props) => {
  const locale = propsLocale ?? i18n.language;
  const article = transformArticle(propsArticle, locale);
  const scripts = getArticleScripts(article);
  const contentUrl = resource?.path
    ? `${config.ndlaFrontendDomain}${resource.path}`
    : undefined;
  return (
    <OneColumn>
      <Helmet>
        <title>{`NDLA | ${article.title}`}</title>
        <meta name="robots" content="noindex" />
        {scripts.map(script => (
          <script
            key={script.src}
            src={script.src}
            type={script.type}
            async={script.async}
            defer={script.defer}
          />
        ))}
      </Helmet>
      <SocialMediaMetadata
        title={article.title}
        image={article.metaImage}
        description={article.metaDescription}
        locale={locale}
        trackableContent={article}
      />
      <PostResizeMessage />
      <FixDialogPosition />
      <Article article={article} locale={locale} {...getArticleProps(resource)}>
        <CreatedBy
          name={t('createdBy.content')}
          description={t('createdBy.text')}
          url={contentUrl}
        />
      </Article>
    </OneColumn>
  );
};

IframeArticlePage.getDocumentTitle = ({ article }: Pick<Props, 'article'>) => {
  return article.id ? `NDLA | ${article.title}` : '';
};

IframeArticlePage.getDimensions = ({ resource, article }: Props) => {
  const articleProps = getArticleProps(resource?.id ? resource : undefined);
  return getAllDimensions({ article }, articleProps.label, true);
};

IframeArticlePage.willTrackPageView = (
  trackPageView: (props: Props) => void,
  currentProps: Props,
) => {
  const { article } = currentProps;
  if (article?.id) {
    trackPageView(currentProps);
  }
};

export default withTranslation()(withTracker(IframeArticlePage));
