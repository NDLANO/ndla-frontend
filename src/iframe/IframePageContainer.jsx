/**
 * Copyright (c) 2020-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import IframePageWrapper from './IframePageWrapper';
import IframePage from './IframePage';

const IframePageContainer = ({
  basename,
  status,
  locale,
  resourceId,
  location,
  articleId,
  isOembed,
  isTopicArticle,
}) => (
  <IframePageWrapper basename={basename} locale={locale}>
    <IframePage
      status={status}
      locale={locale}
      resourceId={resourceId}
      location={location}
      articleId={articleId}
      isOembed={isOembed}
      isTopicArticle={isTopicArticle}
    />
  </IframePageWrapper>
);

IframePageContainer.propTypes = {
  basename: PropTypes.string,
  locale: PropTypes.shape({
    abbreviation: PropTypes.string.isRequired,
    messages: PropTypes.object.isRequired,
  }).isRequired,
  articleId: PropTypes.string,
  resourceId: PropTypes.string,
  status: PropTypes.oneOf(['success', 'error']),
  location: PropTypes.shape({
    pathname: PropTypes.string,
  }),
  isOembed: PropTypes.string,
  isTopicArticle: PropTypes.bool,
};

export default IframePageContainer;
