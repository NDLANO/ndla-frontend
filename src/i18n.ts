/**
 * Copyright (c) 2018-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { formatNestedMessages } from '@ndla/i18n';
// @ts-ignore
import { messagesNB, messagesEN, messagesNN } from '@ndla/ui';
import additionalMessagesNB from './messages/messagesNB';
import additionalMessagesNN from './messages/messagesNN';
import additionalMessagesEN from './messages/messagesEN';
import { LocaleType } from './interfaces';

type LocaleObject = {
  name: string;
  abbreviation: LocaleType;
  messages: {
    [key: string]: string;
  };
};

const NB: LocaleObject = {
  name: 'BokmÃ¥l',
  abbreviation: 'nb',
  messages: formatNestedMessages({ ...messagesNB, ...additionalMessagesNB }),
};
const NN: LocaleObject = {
  name: 'Nynorsk',
  abbreviation: 'nn',
  messages: formatNestedMessages({ ...messagesNN, ...additionalMessagesNN }),
};
const EN: LocaleObject = {
  name: 'English',
  abbreviation: 'en',
  messages: formatNestedMessages({ ...messagesEN, ...additionalMessagesEN }),
};

export const appLocales = [NB, NN, EN];
export const preferredLocales = [NB, NN];

export const getLocaleObject = (localeAbbreviation?: string): LocaleObject => {
  const locale = appLocales.find(l => l.abbreviation === localeAbbreviation);

  return locale || NB; // defaults to NB
};

export const isValidLocale = (localeAbbreviation?: string): boolean =>
  appLocales.find(l => l.abbreviation === localeAbbreviation) !== undefined;

export const getHtmlLang = (localeAbbreviation?: string): string => {
  const locale = appLocales.find(l => l.abbreviation === localeAbbreviation);
  return locale ? locale.abbreviation : 'nb'; // Defaults to nb if not found
};

interface RetType extends LocaleObject {
  basepath: string;
  basename: string;
}

export const getLocaleInfoFromPath = (path: string): RetType => {
  const paths = path.split('/');
  const basename = paths[1] && isValidLocale(paths[1]) ? paths[1] : '';
  const basepath = basename ? path.replace(`/${basename}`, '') : path;
  return {
    basepath,
    basename,
    ...getLocaleObject(basename),
  };
};
