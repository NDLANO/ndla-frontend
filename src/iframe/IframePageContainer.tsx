/**
 * Copyright (c) 2020-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { Helmet } from "react-helmet-async";
import { PageContainer } from "@ndla/ui";
import IframePage from "./IframePage";
import { BaseNameProvider } from "../components/BaseNameContext";
import Scripts from "../components/Scripts/Scripts";
import { isValidLocale } from "../i18n";
import { LocaleType } from "../interfaces";

interface Props {
  locale?: LocaleType;
  basename?: string;
  articleId?: string;
  taxonomyId?: string;
  status?: "success" | "error";
  isOembed?: string;
}
const IframePageContainer = ({ basename, status, taxonomyId, articleId, isOembed, locale }: Props) => {
  return (
    <BaseNameProvider value={isValidLocale(basename) ? basename : ""}>
      <PageContainer>
        <Scripts />
        <Helmet htmlAttributes={{ lang: locale === "nb" ? "no" : locale }} />
        <IframePage status={status} taxonomyId={taxonomyId} articleId={articleId} isOembed={isOembed} />
      </PageContainer>
    </BaseNameProvider>
  );
};

export default IframePageContainer;
