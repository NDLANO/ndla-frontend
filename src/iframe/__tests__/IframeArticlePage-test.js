/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @jest-environment jsdom
 */

import React from 'react';
import nock from 'nock';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';
import IntlProvider from '@ndla/i18n';
import IframePageContainer from '../IframePageContainer';
import { getLocaleObject } from '../../i18n';
import IframeArticlePage, { fetchResourceId } from '../IframeArticlePage';

expect.addSnapshotSerializer(serializer);

test('IframeArticlePage with article renderers correctly', () => {
  nock('http://ndla-api')
    .get('/taxonomy/v1/resources/urn:resource:1/?language=nb')
    .reply(200, {
      id: 'urn:resource:1',
      title: 'Ressurs',
    });

  const locale = getLocaleObject('nb');
  const article = {
    content:
      '<section><p>Dersom du leser de ulike partiprogrammene, ser du fort at partiene har ulike svar både på hva som er viktige utfordringer, og på hvordan de skal løses.</p></section>',
    created: '2018-01-09T18:40:03Z',
    introduction:
      'Politiske skillelinjer, eller konfliktlinjer, er varige og grunnleggende motsetninger i samfunnet og blant velgerne. Du synes kanskje det er vanskelig å se forskjell på de politiske partiene – det er du i så fall ikke alene om!',
    metaDescription: 'Politiske skillelinjer, eller konfliktlinjer',
    metaData: {
      footnotes: '',
    },
    copyright: {
      license: {
        license: 'by-sa',
        description: 'Creative Commons Attribution-ShareAlike 2.0 Generic',
        url: 'https://creativecommons.org/licenses/by-sa/2.0/',
      },
      creators: [
        {
          type: 'Writer',
          name: 'Someone',
        },
      ],
      processors: [],
      rightsholders: [],
    },
    oldNdlaUrl: '//red.ndla.no/node/168554',
    requiredLibraries: [],
    title: 'Politiske skillelinjer',
    updated: '2018-01-09T18:43:48Z',
    published: '2018-01-09T18:43:48Z',
    supportedLanguages: ['nb'],
  };
  const component = renderer.create(
    <IntlProvider locale={locale.abbreviation} messages={locale.messages}>
      <IframeArticlePage
        locale={locale.abbreviation}
        location={{ pathname: '/article-iframe/urn:resource:1/128' }}
        resource={{
          id: 'urn:resource:1',
          name: 'Ressurs',
          path: '/subject:1/resource:1',
          article,
          resourceTypes: [],
        }}
        article={article}
      />
    </IntlProvider>,
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('IframePage with article displays error message on status === error', () => {
  const component = renderer.create(
    <IframePageContainer
      location={{ pathname: '/article-iframe/333' }}
      locale={getLocaleObject('nb')}
      status="error"
    />,
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('fetchResourceId fetches correct resource id from path', () => {
  const url =
    'https://ndla.no/article-iframe/urn:resource:670ac97d-1d4d-4515-9554-07e0870e66aa/24835?removeRelatedContent=true';
  expect(fetchResourceId({ location: { pathname: url } })).toMatch(
    'urn:resource:670ac97d-1d4d-4515-9554-07e0870e66aa',
  );

  const urlWithLang =
    'https://ndla.no/article-iframe/nb/urn:resource:670ac97d-1d4d-4515-9554-07e0870e66aa/24835?removeRelatedContent=true';
  expect(fetchResourceId({ location: { pathname: urlWithLang } })).toMatch(
    'urn:resource:670ac97d-1d4d-4515-9554-07e0870e66aa',
  );
});
