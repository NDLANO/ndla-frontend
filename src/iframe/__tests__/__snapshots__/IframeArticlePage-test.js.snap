// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IframeArticlePage with article renderers correctly 1`] = `
.emotion-0 {
  display: inline-block;
  color: #ffffff;
  background-color: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  padding: 4px 13px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.625;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 700;
  -webkit-transition: all .2s cubic-bezier(0.17,0.04,0.03,0.94);
  transition: all .2s cubic-bezier(0.17,0.04,0.03,0.94);
  box-shadow: none;
  text-align: center;
  font-size: 12px;
  font-size: 0.6666666666666666rem;
  line-height: 15px;
  background-color: #deebf6;
  border: none;
  color: #20588f;
  font-weight: 600;
  position: absolute;
  z-index: 9;
  right: 0;
  top: 0;
  display: none;
}

.emotion-0:hover,
.emotion-0:focus {
  color: white;
  background-color: #184673;
  border: 2px solid rgba(32,88,143,0);
  -webkit-transform: translateY(1px) translateX(1px);
  -ms-transform: translateY(1px) translateX(1px);
  transform: translateY(1px) translateX(1px);
}

.emotion-0:active {
  -webkit-transform: translateY(2px) translateX(2px);
  -ms-transform: translateY(2px) translateX(2px);
  transform: translateY(2px) translateX(2px);
}

.emotion-0[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
  -webkit-transform: translateY(0) translateX(0);
  -ms-transform: translateY(0) translateX(0);
  transform: translateY(0) translateX(0);
}

.emotion-0:focus {
  box-shadow: 0 0 2px #20588f;
}

.emotion-0:hover,
.emotion-0:focus,
.emotion-0:active {
  border: none;
  background: #20588f;
}

@media (min-width:61.3125em) {
  .emotion-0 {
    display: block;
  }
}

.emotion-2 {
  display: inline-block;
  color: #ffffff;
  background-color: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  padding: 4px 13px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.625;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 700;
  -webkit-transition: all .2s cubic-bezier(0.17,0.04,0.03,0.94);
  transition: all .2s cubic-bezier(0.17,0.04,0.03,0.94);
  box-shadow: none;
  text-align: center;
  -webkit-transition: background-color none;
  transition: background-color none;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
  font-size: inherit;
  line-height: inherit;
  color: #20588f;
  box-shadow: inset 0 -1px;
}

.emotion-2:hover,
.emotion-2:focus {
  color: white;
  background-color: #184673;
  border: 2px solid rgba(32,88,143,0);
  -webkit-transform: translateY(1px) translateX(1px);
  -ms-transform: translateY(1px) translateX(1px);
  transform: translateY(1px) translateX(1px);
}

.emotion-2:active {
  -webkit-transform: translateY(2px) translateX(2px);
  -ms-transform: translateY(2px) translateX(2px);
  transform: translateY(2px) translateX(2px);
}

.emotion-2[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
  -webkit-transform: translateY(0) translateX(0);
  -ms-transform: translateY(0) translateX(0);
  transform: translateY(0) translateX(0);
}

.emotion-2:focus {
  box-shadow: 0 0 2px #20588f;
}

.emotion-2:hover,
.emotion-2:active,
.emotion-2:disabled,
.emotion-2:focus {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  border: none;
}

.emotion-2:focus {
  outline-width: medium;
}

.emotion-2:hover,
.emotion-2:focus {
  box-shadow: none;
}

.emotion-4 {
  display: inline-block;
  color: #ffffff;
  background-color: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  padding: 4px 13px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.625;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 700;
  -webkit-transition: all .2s cubic-bezier(0.17,0.04,0.03,0.94);
  transition: all .2s cubic-bezier(0.17,0.04,0.03,0.94);
  box-shadow: none;
  text-align: center;
  font-size: 12px;
  font-size: 0.6666666666666666rem;
  line-height: 15px;
  background-color: #deebf6;
  border: none;
  color: #20588f;
  font-weight: 600;
}

.emotion-4:hover,
.emotion-4:focus {
  color: white;
  background-color: #184673;
  border: 2px solid rgba(32,88,143,0);
  -webkit-transform: translateY(1px) translateX(1px);
  -ms-transform: translateY(1px) translateX(1px);
  transform: translateY(1px) translateX(1px);
}

.emotion-4:active {
  -webkit-transform: translateY(2px) translateX(2px);
  -ms-transform: translateY(2px) translateX(2px);
  transform: translateY(2px) translateX(2px);
}

.emotion-4[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
  -webkit-transform: translateY(0) translateX(0);
  -ms-transform: translateY(0) translateX(0);
  transform: translateY(0) translateX(0);
}

.emotion-4:focus {
  box-shadow: 0 0 2px #20588f;
}

.emotion-4:hover,
.emotion-4:focus,
.emotion-4:active {
  border: none;
  background: #20588f;
}

@media (min-width:61.3125em) {
  .emotion-4 {
    display: none;
  }
}

.emotion-6 {
  border: calc(52px / 6.5) solid rgba(0,0,0,0.1);
  border-bottom-color: #20588f;
  border-radius: 50%;
  -webkit-animation: spinnerAnimation 0.7s linear infinite;
  animation: spinnerAnimation 0.7s linear infinite;
  height: 52px;
  width: 52px;
  display: block;
  margin: 26px auto;
}

<div
  className="o-container"
>
  <div
    className="o-wrapper undefined"
  >
    <article
      className="c-article c-article--clean c-article--iframe"
    >
      <section
        className="u-4/6@desktop u-push-1/6@desktop u-10/12@tablet u-push-1/12@tablet"
      >
        <div
          className="c-article__header"
        >
          <button
            className="emotion-0 emotion-1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Vis kompetansemål
          </button>
          <div
            className="c-article__title"
          >
            <h1>
              Politiske skillelinjer
            </h1>
          </div>
          <p
            className="article_introduction"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Politiske skillelinjer, eller konfliktlinjer, er varige og grunnleggende motsetninger i samfunnet og blant velgerne. Du synes kanskje det er vanskelig å se forskjell på de politiske partiene – det er du i så fall ikke alene om!",
              }
            }
          />
          <div
            className="c-article-byline"
          >
            <span
              className="c-article-byline__flex"
            >
              <span
                className="c-article-byline__icon"
              >
                <svg
                  className="c-icon"
                  data-license="Apache License 2.0"
                  data-source="Material Design"
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "color": undefined,
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 48 48"
                  width="1em"
                >
                  <title>
                    User
                  </title>
                  <g>
                    <path
                      d="M24 24c4.42 0 8-3.59 8-8 0-4.42-3.58-8-8-8s-8 3.58-8 8c0 4.41 3.58 8 8 8zm0 4c-5.33 0-16 2.67-16 8v4h32v-4c0-5.33-10.67-8-16-8z"
                    />
                  </g>
                </svg>
              </span>
              <span
                className="c-article-byline__authors"
              >
                Someone
              </span>
              <span>
                 (CC BY-SA)
              </span>
            </span>
            <span
              className="c-article-byline__flex"
            >
              <span
                className="c-article-byline__icon"
              >
                <svg
                  className="c-icon"
                  data-license="Apache License 2.0"
                  data-source="Material Design"
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "color": undefined,
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 48 48"
                  width="1em"
                >
                  <title>
                    Time
                  </title>
                  <g>
                    <path
                      d="M23.99 4C12.94 4 4 12.95 4 24s8.94 20 19.99 20C35.04 44 44 35.05 44 24S35.04 4 23.99 4zM24 40c-8.84 0-16-7.16-16-16S15.16 8 24 8s16 7.16 16 16-7.16 16-16 16zm1-26h-3v12l10.49 6.3L34 29.84l-9-5.34z"
                    />
                  </g>
                </svg>
              </span>
              <span
                className="c-article-byline__date"
              >
                Sist oppdatert
                 
                09.01.2018
              </span>
            </span>
            <span
              className="c-article-byline__flex"
            >
              <button
                className="emotion-2 emotion-1"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Bruk innhold
              </button>
            </span>
          </div>
          <button
            className="emotion-4 emotion-1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Vis kompetansemål
          </button>
          <div
            className="emotion-6 emotion-7"
            size="large"
          />
        </div>
      </section>
      <section
        className="u-4/6@desktop u-push-1/6@desktop u-10/12@tablet u-push-1/12@tablet"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<section><p>Dersom du leser de ulike partiprogrammene, ser du fort at partiene har ulike svar både på hva som er viktige utfordringer, og på hvordan de skal løses.</p></section>",
            }
          }
        />
      </section>
      <section
        className="u-4/6@desktop u-push-1/6@desktop u-10/12@tablet u-push-1/12@tablet"
      />
      <section
        className="u-10/12@desktop u-push-1/12@desktop u-10/12@tablet u-push-1/12@tablet"
      >
        <a
          className="article-old-ndla-link"
          href="//red.ndla.no/node/168554"
          rel="noopener noreferrer"
          target="_blank"
        >
          Gå til orginal artikkel
        </a>
      </section>
    </article>
  </div>
</div>
`;

exports[`IframePage with article displays error message on status === error 1`] = `
.emotion-4 {
  text-align: center;
}

.emotion-4 a {
  color: #20588f;
}

.emotion-4 h1 {
  margin-top: 0;
}

.emotion-0 {
  margin-bottom: 26px;
}

@media (min-width:37.5625em) {
  .emotion-0 {
    margin-top: 52px;
  }
}

.emotion-2 {
  margin-bottom: 26px;
}

@media (min-width:37.5625em) {
  .emotion-2 {
    margin-bottom: 52px;
  }
}

<div
  className="o-container"
>
  <div
    className="o-wrapper o-wrapper--clear undefined"
  >
    <article
      className="emotion-4 emotion-5"
    >
      <img
        alt="Ops, noe gikk galt"
        className="emotion-0 emotion-1"
        src="/static/oops.gif"
      />
      <h1>
        Ops, noe gikk galt
      </h1>
      <p
        className="emotion-2 emotion-3"
      >
        Beklager, en feil oppsto.
      </p>
    </article>
  </div>
</div>
`;
